src/frontends/lean/definition_cmds.cpp

```diff
@@ -314,9 +315,18 @@ static certified_declaration check(parser & p, environment const & env, name con
                     << "type checking time for " << c_name << " "
                     << std::fixed << std::setprecision(5) << duration << " secs\n";
                 msg.report();
-            });
+        });
+
+        if (linting_enabled(p.get_options())) {
+            lint_declaration(env, d);
+        }
+
         return ::lean::check(env, d);
     } else {
+        if (linting_enabled(p.get_options())) {
+            lint_declaration(env, d);
+        }
```

Comment:
I agree.  Aside from that, the linter function should also be invoked on lemmas, and probably on examples as well.